doc/releases/migration-guide-3.5.rst

```diff
@@ -19,40 +22,33 @@ Required changes
   detect if you pass the function memory not belonging to the memory blocks in
   the slab.
 
+* :c:macro:`CONTAINER_OF` now performs type checking, this was very commonly
+  misused to obtain user structure from :c:struct:`k_work` pointers without
+  passing from :c:struct:`k_work_delayable`. This would now result in a build
+  error and have to be done correctly using
+  :c:func:`k_work_delayable_from_work`.
+
+Configuration System (Kconfig)
```

Comment:
This sounds a bit unclear, if feels like it should contain changes of the Kconfig system itself, plus many other areas contains config changes

How about having a "Core OS" are for project wide changes including config options like the bootloader one, and moving the nct38xx under "Device Drivers"?